pkg/config/config.go

```diff
@@ -14,12 +15,17 @@ import (
 const (
 	configEnvName  = "ODOCONFIG"
 	configFileName = "odo"
+	// ConfigPrefixDir The default value for config `prefix`.
+	// When set to $DIR, the directory name of component dir(or current-dir if not available) will be used as prefix for app and component names
+	ConfigPrefixDir = "$DIR"
```

Comment:
can we use something else for the value that doesn't also represent env variable?
This will be tricky to enter on the command line as it will have to be escaped.
Also, it is a little bit misleading. For git and binary component it will be something else than name of the directory.

something like AUTOMATIC might be a more appropriate name. What do you think?

Do we even need this, can we just assume that if the prefix is not set than it is this?

Reply:
Ack

Reply:
> Do we even need this, can we just assume that if the prefix is not set than it is this?

@kadel Without a reserved keyword, if user wants to reset to default after configuring it to a non-default value, he would only be left with 1 route => remove the configurable manually from conf file(and then default will be picked up from within code). But, reserving a word takes away a possible value for the configurable.. Not sure what is the best way.. Please suggest

Reply:
How about the empty string? `odo utils config set NamePrefix ""`